extension/ts/services/google/googleFetch.ts

```diff
@@ -1,29 +1,24 @@
-import Auth from "./googleAuth";
+import {invokeWorker} from "../../workers/invoker";
+import {WorkerKind} from "../../workers/types";
+import {showToast, ToastType} from "../../ui/toast";
 
 const handleError = (res: Response) => {
-	if (res.status >= 400) {
-		Auth.clearAPICredentials();
-		alert(
-			"Something went wrong when trying to process that action. Please try again and ensure the inputted API key/ Client ID credentials are correct."
-		);
+	if (!res.ok) {
+		showToast("Could not complete action", ToastType.ERROR);
 	}
 };
 
 type GoogleFetch = <T>(...args: Parameters<typeof fetch>) => Promise<T | null>;
 
 export const googleFetch: GoogleFetch = async (input: URL, init?: RequestInit) => {
-	const credentials = Auth.getAPICredentials();
 	const url = new URL(input);
 	const params = new URLSearchParams(url.search);
-	params.append("key", credentials.apiKey);
 	url.search = params.toString();
 	init ??= {};
 	init.headers ??= {};
 
 	if (init.method !== "GET") {
-		Auth.authorize();
-		const localStorageOAuth = Auth.getOAuthCredentials();
-		const token = localStorageOAuth.access_token;
+		const token = await invokeWorker(WorkerKind.Authorize, null);
 		init = {...init, headers: {...init.headers, Authorization: `Bearer ${token}`}};
```

Comment:
nit: i think this always gets run lmao -- in the callsites we're making a GET request, we're not passing in `init.method = "GET`

that's why get requests are still working without an API key -- i think it's adding the bearer token to the request

we might as well remove the `init.method !== GET` guard (?)

Reply:
WAIT NO i misunderstood

Reply:
so we should have been checking for `undefined` and `"GET"`? OK I will test it out in the morning

i am on gamer sleep cycles rn

Reply:
UGH ok I added
```typescript
init.method ??= "GET"
```
and it broke the whole thing. Yeah I'll remove the condition

Reply:
LMAO we had another bug there
It ends in `fetch(init)` instead of `fetch(url)`, so we never were passing in the api key